service_config.go

```diff
@@ -285,10 +285,6 @@ func convertRetryPolicy(jrp *jsonRetryPolicy) (p *internalserviceconfig.RetryPol
 		BackoffMultiplier:    jrp.BackoffMultiplier,
 		RetryableStatusCodes: make(map[codes.Code]bool),
 	}
-	if rp.MaxAttempts > 5 {
```

Comment:
Is it possible to keep the relevant logic here (where all the retry code is) by parameterizing the "max max attempts" setting, instead of moving it into the RPC path?

Reply:
I can certainly try :)

Reply:
@dfawley Looking at all the places `parseServiceConfig` is called, If I parameterize this, we end up passing the max call setting around from dial options to the dns resolver, load balancer config, as well as here in service config. It would get messy real fast. Do you want me to do this?

Reply:
That doesn't sound right to me.  Resolvers should be calling `ParseConfig` on the `ClientConn`, and that should be adding the setting in at that time.  Maybe I'm missing something though?  Is it tests that are the problem?

Reply:
You are mostly right. There are a couple other places (aside from tests) where it is called as `internal.ParseServiceConfig` where it doesn't have access to `ClientConn`, but in those places `maxAttempts` seems inconsequential.

Reply:
I've updated as you've suggested.